test/vamp-client/tst_PluginStub.cpp

```diff
@@ -0,0 +1,153 @@
+    finish(PluginStub* /*plugin*/) override
+    {
+        return {};
+    }
+    
+    void
+    reset(PluginStub* /*plugin*/, PluginConfiguration /*config*/) override
+    {}
+private:
+    PluginStaticData m_staticData;
+};
+
+
+TEST_CASE("Init plugin with parameter dependent preferred framing sizes") {
+    const std::size_t initialBlockSize = 1024;
+    const std::size_t initialStepSize = 512;
+    PluginConfiguration defaultConfig;
+    defaultConfig.channelCount = 1;
+    defaultConfig.framing.blockSize = initialBlockSize;
+    defaultConfig.framing.stepSize = initialStepSize;
+    defaultConfig.parameterValues = {{"framing-scale", 1.0}};
+    
+    Vamp::PluginBase::ParameterDescriptor stubParam;
+    stubParam.identifier = "framing-scale";
+    stubParam.name = "Framing Scale Factor";
+    stubParam.description = "Scales the preferred framing sizes";
+    stubParam.maxValue = 2.0; 
+    
+    PluginStaticData staticData;
+    staticData.pluginKey = "stub";
+    staticData.basic = {"stub:param-init", "Stub", "Testing init"};
+    staticData.maker = "Lucas Thompson";
+    staticData.copyright = "GPL";
+    staticData.pluginVersion = 1;
+    staticData.category = {"Test"};
+    staticData.minChannelCount = 1;
+    staticData.maxChannelCount = 1;
+    staticData.parameters = {stubParam};
+    staticData.inputDomain = Vamp::Plugin::InputDomain::TimeDomain;
+    staticData.basicOutputInfo = {{"output", "NA", "Not real"}};
+
+    StubClient stub {staticData};
+    
+    PluginStub vampPiperAdapter {
+        &stub, 
+        "stub", // plugin key
+        44100.0, // sample rate
+        0, // adapter flags, don't care here
+        staticData, 
+        defaultConfig 
+    };
+    
+    const auto initWithPreferredFraming = [&]() -> bool {
+        return vampPiperAdapter.initialise(
+            1, 
+            vampPiperAdapter.getPreferredStepSize(), 
+            vampPiperAdapter.getPreferredBlockSize()
+        );  
+    };
+    
+    SECTION("Initialises with default parameters")
+    {
+        REQUIRE( initWithPreferredFraming() );   
+    }
+    
+    SECTION("Fails to init when changing framing influencing parameter")
+    {
+        const float scalingFactor = 2.0;
+        vampPiperAdapter.setParameter("framing-scale", scalingFactor);
+        REQUIRE( initWithPreferredFraming() == false );
+        const float configuredStepSize = vampPiperAdapter.getPreferredStepSize();
+        const float configuredBlockSize = vampPiperAdapter.getPreferredBlockSize();
+        REQUIRE( configuredStepSize == initialStepSize * scalingFactor );
+        REQUIRE( configuredBlockSize == initialBlockSize * scalingFactor );
+    }
+    
+    SECTION("Cannot process after a failed init call (due to framing)")
+    {
+        const float scalingFactor = 2.0;
+        vampPiperAdapter.setParameter("framing-scale", scalingFactor);
+        REQUIRE( initWithPreferredFraming() == false );
+        REQUIRE_THROWS( vampPiperAdapter.process(nullptr, {}) );
```

Comment:
I don't like passing null here, because it's theoretically possible that the adapter could be throwing simply because it received a null pointer